src/qibolab/_core/sequence.py

```diff
@@ -262,3 +263,51 @@ def by_channel(self) -> dict[ChannelId, list[PulseLike]]:
             seqs[ch].append(pulse)
 
         return seqs
+
+    def to_vzs(self) -> "PulseSequence":
+        """Split relative phases to :class:`VirtualZ` elements.
+
+        The basic formula just relies on the composition of Pauli matrices' exponential
+        and it is available from many sources. One of those could be MM p. 626.
+
+        .. todo::
+
+            Add MM as a proper reference, such that it ends up in the bibliography
+        """
+        return PulseSequence(
+            [
+                el
+                for els in (
+                    [(ch, ev)]
+                    if not isinstance(ev, Pulse) or np.isclose(ev.relative_phase, 0)
+                    else [
+                        (ch, VirtualZ(phase=ev.relative_phase)),
+                        (ch, ev.model_copy(update={"relative_phase": 0})),
+                        (ch, VirtualZ(phase=-ev.relative_phase)),
+                    ]
+                    for ch, ev in self
+                )
+                for el in els
+            ]
+        )
+
+    def to_relative_phases(self) -> "PulseSequence":
+        seq = PulseSequence()
+        phases = defaultdict(float)
+
+        for ch, ev in self:
+            if isinstance(ev, VirtualZ):
+                phases[ch] += ev.phase
+            elif isinstance(ev, Pulse):
+                seq.append(
+                    (
+                        ch,
+                        ev.model_copy(
+                            update={"relative_phase": ev.relative_phase + phases[ch]}
```

Comment:
This is the only reason why I'm not completely satisfied with this PR: to my understanding it should have been 
```suggestion
                            update={"relative_phase": ev.relative_phase - phases[ch]}
```

<img width="927" height="223" alt="Image" src="https://github.com/user-attachments/assets/588dbca1-cb61-4140-8f31-09da9b9b38e3" />

Instead, I'm getting as if the phase is minus the rotation angle. At least for the emulator.
Which could be. But I'm just a bit puzzled. And I'm worried it could be not consistent in the electronics (that's why I'm waiting to test it also on hardware).